packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -203,6 +204,20 @@ class NavigationBar extends StatelessWidget {
   /// is also null, the default is 80.
   final double? height;
 
+  /// The adjustment on the bottom padding of the [SafeArea].
+  ///
+  /// [NavigationBar] respects the [SafeArea], so its height includes its own
+  /// default height plus the bottom padding of the [SafeArea]. To display more
+  /// content in the main body, [NavigationBar.height] can be updated to a smaller
+  /// value, but this may cause an overlap with the bottom [SafeArea], making
+  /// part of the navigation bar untappable. In this case, this property can be
```

Comment:
It feels like in this case we should be asserting instead of putting tappable elements in the safe area. Won't some devices disallow tapping if it is in one of the system gesture areas? Like the bottom of the iPhone?